db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/check_ssh.rb

```diff
@@ -11,13 +13,20 @@ def remove_deployment_master(hosts, master_ip)
 
 def check_ssh
   $evm.log(:info, "**************** #{$evm.root['ae_state']} ****************")
-
   deployment_hosts = $evm.root['masters'] + $evm.root['nodes']
   deployment_hosts = remove_deployment_master(deployment_hosts, $evm.root['deployment_master'])
-
   begin
-    Net::SSH.start($evm.root['deployment_master'], $evm.root['user'], :paranoid => false, :forward_agent => true,
-                   :key_data => $evm.root['private_key']) do |ssh|
+    agent_socket = "/tmp/ssh_manageiq/ssh_manageiq_#{$evm.root['automation_task'].automation_request.options[:attrs][:deployment_id]}"
+    FileUtils.mkdir_p '/tmp/ssh_manageiq'
+    agent_details = `ssh-agent -a #{agent_socket}`
+    sock = agent_details.split("=")[1].split(" ")[0].chop
+    pid = agent_details.split("=")[2].split(" ")[0].chop
+    deployment = $evm.root['deployment']
+    $evm.root['agent_pid'] = pid
+    $evm.root['agent_socket'] = agent_socket
+    deployment.customize({:ssh_details => {:socket => sock, :pid => pid}})
```

Comment:
@alongoldboim @simon3z 
why do we need to save the agent transient data in the DB ?